### PR TITLE
Support non-default and differing primary key fields.

### PR DESCRIPTION
**Pull request by bmomberger-bitovi**
Previously, primary keys that were not "id" might cause failures, and using primary keys that were not the same on both sides of an association did cause failures.  This PR refactors all of the handlers for create, bulk create, and update to allow for primary keys to be any single key.  Composite primary keys are still not supported, but that's not a requirement at the moment.

* Also add transaction to a few queries where it was missing; adding it ensures that all queries run in the correct order and gets pre-commit data.

* Also enable declarations in tsconfig, so consuming apps can use strict typing.

### CODE REVIEW

File: src/sequelize/associations/sequelize.post.ts [hunk 1]
```diff
@@ -113,12 +113,9 @@ export const handleCreateMany = async (
   transaction: Transaction,
   primaryKey = "id",
 ): Promise<void> => {
-  const modelInstance = await sequelize.models[model.name].findByPk(
-    model[primaryKey],
-    {
-      transaction,
-    },
-  );
+  const modelInstance = await sequelize.models[model.name].findByPk(model.id, {
```
Comment (roy-coder) [inline src/sequelize/associations/sequelize.post.ts:121]: Is this change intended to disallow `primaryKey` different than `id`?

Comment (bmomberger-bitovi) [reply src/sequelize/associations/sequelize.post.ts:121]: Actually the opposite.  `model` here is not a full model but a constructed object with an `id` field that contains the primary key value[s].  Trying to read `primaryKey` off of `model` will be undefined  However, you did point out that I need to do the same for bulkCreate below

Comment (roy-coder) [reply src/sequelize/associations/sequelize.post.ts:121]: Then probably in `sequelize.patch.ts` as well

Comment (bmomberger-bitovi) [reply src/sequelize/associations/sequelize.post.ts:121]: Yup and supporting non-default primary keys and PKs that are different on each side has taken me all day, but now it's complete.

**Review by roy-coder — APPROVED** [review]
